services/telemetry/telemetry.go

```diff
@@ -153,20 +204,44 @@ func (s *Service) sendOnce(ctx context.Context) error {
 		return err
 	}
 
-	payload := s.makePayload(settings.Telemetry.UUID)
-	return s.sendRequest(ctx, payload)
+	var wg errgroup.Group
+
+	wg.Go(func() error {
+		payload := s.makeV1Payload(settings.Telemetry.UUID)
+		return s.sendV1Request(sendOnceCtx, payload)
+	})
+
+	wg.Go(func() error {
+		req, err := s.makeV2Payload(settings.Telemetry.UUID)
+		if err != nil {
+			return err
+		}
+
+		if err = s.sendV2Request(sendOnceCtx, req); err != nil {
+			s.l.Debugf("Fail to send telemetry v2, add request to retry queue: %s", req)
+			go s.runWaitAndRetry(ctx, req)
```

Comment:
I don't think we should create another grouting there – we are already in `wg.Go` call. It is totally fine for `sendOnce` to block for several hours.

So it looks like we can have all retries logic right there – to keep all concurrency and synchronisation close.

Reply:
If we block on `sendOnce` that means that no more metrics will be sent until failed request successfully retried or expired. For example, a single invalid `bad request`, could hang whole metrics delivery for 20 hours. But if we expect only network/server problems, then yes, we can eliminate new goroutine spawning and block `sendOnce`

So the solution depends on our expectations from that mechanism.

Reply:
> If we block on sendOnce that means that no more metrics will be sent until failed request successfully retried or expired. For example, a single invalid bad request, could hang whole metrics delivery for 20 hours.

But we send uptime event (not metrics – there are no metrics) every 24 hours. That's intentional.

Reply:
What we can do is to make 24 hour a constant too in this package, and then add a test that checks that constants are in sync with each other: `delay < retryInterval * retryCount`.